### ISSUE TITLE
ComboBox has issues announcing number of items in section in TalkBack/iOS voiceover 

### ISSUE THREAD

**Issue by LFDanLu** [adobe/react-spectrum#2578]
# 🐛 Bug Report 

When a user navigates into a ComboBox section via TalkBack, TalkBack announces the item's position correctly but not the number of items in the section, stating "item 3 of 0". iOS VoiceOver doesn't announce the item's position or number of items in the section at all. Seems to be an issue of screen reader support perhaps, reproduces with sandbox below. Note that this is a problem with a ComboBox w/ sections only 

Experiments with `aria-posinset` and `aria-setsize`:
https://codesandbox.io/s/shy-glitter-yvs67?file=/src/index.js
Note that the 4th list announces the number of items correctly in TalkBack (states 8 due to the max posinset being 8), but once the group is introduced the announcement states "item 5 of 0" but will state 8 items at the end. iOS VoiceOver doesn't announce the position or number of items at all.

## 🤔 Expected Behavior

TalkBack should announce the number of items in the section. iOS VoiceOver should announce the number of items in the section + position of items

## 😯 Current Behavior

TalkBack announces "item 3 of 0", should be "item 3 of [num items in section]"
iOS VoiceOver should announce something

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   |. 3.14.1
| Browser          |
| Operating System |

## 🧢 Your Company/Team

RSP
